api/APIv2.md

```diff
@@ -0,0 +1,435 @@
+
+A Device MUST generate its own unique Device certificate and key. This specification does NOT define the mechanism by which a Device is to do so, as it is implementation-dependent. It is RECOMMENDED that a Device use hardware security, where possible, such as TPM or TEE, to protect the Device private key.
+
+A Device MUST have a method of knowing or discovering its Controller URL. This specification does NOT define the mechanism by which a Device is to do so, as it is implementation-dependent.
+
+A Device MUST maintain and persist its own current state such that it survives reboots and process restarts. This includes UUID and configuration parameters. This specification does NOT define the mechanism by which each Device is to do so.
+
+### Controller
+
+A Controller MUST be available for response to Device requests.
+
+A Controller MUST be accessible to all Devices that need to reach it. If it cannot be accessed due to network limitations, the necessary proxies and other network traversal solutions should be made available. Those solutions are implementation-dependent and are not defined in this specification.
+
+A Controller MUST listen for https on a port. The specific port is not specified in this document, however, it is RECOMMENDED to use the well-known https port of 443 where possible. A Controller MUST expose all of the endpoints listed in this document. A Controller MAY additionally listen on other ports, and MAY expose additional endpoints defined in this document, provided those endpoints do NOT conflict with those defined in this document, and use the "Extensions" provision listed herein.
+
+A Controller MUST generate a UUID for each Device. The UUID MUST be unique across all Devices managed by the Controller. The Controller SHOULD use an approach to generating UUIDs which minimizes the probability of UUID collisions with UUIDs generated by all Controllers everywhere.
+
+## Messages
+
+All messages are defined in subdirectories to the [proto](./proto) directory.
+In general, there is one directory for each API endpoint:
+
+* `register`: The register message sent as part of onboarding a device
+* `config`: The ConfigRequest message sent from the device and the ConfigResponse with EdgeDevConfig message sent from Controller to Device in response.
+* `info`: The ZInfoMsg message sent from Device to Controller when there is a state change for an object (device, app instance, etc
+)
+* `metrics`: The ZMetricMsg message sent from Device to Controller periodically to report on resource usage etc.
+* `logs`: The LogBundle message sent from Device to Controller containing internal device logs.
+* `apps/instances/{app-instance-uuid}/logs`: The LogBundle message sent from Device to Controller containing application console device logs.
```

Comment:
Apps log V2 API is:
```apps/instanceid/%s/logs```
This is what is being used in code today and works.

Reply:
Will fix